loadsbroker/extensions.py

```diff
@@ -587,3 +469,129 @@ def _ssh_exec(self, instance: EC2Instance, cmd: str) -> int:
             if status:
                 logger.error("ssh cmd failed:\n%s", stderr.read())
             return status
+
+
+class Grafana:
+    """Grafana monitor Dashboard for AWS instances"""
+
+    data_source_defaults = dict(
+        type='influxdb',
+        access='proxy',
+        isDefault=True,
+        basicAuth=False
+    )
+
+    def __init__(self, info) -> None:
+        self.info = info
+
+    async def start(self,
+                    collection: EC2Collection,
+                    run_id: str,
+                    options: InfluxDBOptions):
+        data_source = self.data_source_defaults.copy()
+        data_source.update(
+            name="loads-broker InfluxDB Monitor (run_id: {})".format(run_id),
+            url="http://" + join_host_port(options.host, options.port),
+            database=options.database,
+        )
+
+        port = 8080
+        ports = {3000: port}
+
+        cmd = """\
+        apt-get update -qq && \
+        apt-get install -qq -y --no-install-recommends curl && \
+        /etc/init.d/grafana-server start && \
+        until curl "${__LOADS_GRAFANA_URL__}" \
+                   -X POST \
+                   -H "Accept: application/json" \
+                   -H "Content-Type: application/json" \
+                   --data-binary "${__LOADS_GRAFANA_DS_PAYLOAD__}"; do
+            sleep 1
+        done && \
+        /etc/init.d/grafana-server stop && \
+        mkdir "${GF_DASHBOARDS_JSON_PATH}" && \
+        echo "${__LOADS_GRAFANA_DASHBOARD__}" >> \
```

Comment:
Can't we use HTTP calls to load dashboards (http://docs.grafana.org/http_api/dashboard/#create-update-dashboard)?